### PR TITLE
Image usage exporter

### PR DESCRIPTION
**Pull request by DEiselt**
Helm Chart for the image usage exporter requested by the CC+1 image Team

### CODE REVIEW

File: system/infra-monitoring/vendor/image-usage-exporter/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.0.1
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+appVersion: 0.0.1
+
+maintainers:
+  - name: Danny Eiselt (C5280640)
```
Comment (Finkregh) [inline system/infra-monitoring/vendor/image-usage-exporter/Chart.yaml:26]: i'd add the image DL

**Review by Finkregh — CHANGES_REQUESTED** [review]

File: system/infra-monitoring/vendor/image-usage-exporter/Chart.yaml (hunk 1 above)
Comment (DEiselt) [reply system/infra-monitoring/vendor/image-usage-exporter/Chart.yaml]: Done

File: system/infra-monitoring/requirements.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-dependencies:
+git dependencies:
```
Comment (stefanhipfel) [inline system/infra-monitoring/requirements.lock:1]: not sure why this change?

Comment (DEiselt) [reply system/infra-monitoring/requirements.lock]: Probably due to git conflicts / rebasing. Thank you for pointing this out.

**Review by stefanhipfel — APPROVED** [review]

**Comment by Finkregh** [comment]
`[ERROR] templates/: template: image-usage-exporter/templates/deployment.yaml:33:29: executing "image-usage-exporter/templates/deployment.yaml" at <.Values.global.datahub.user>: nil pointer evaluating interface {}.datahub`

any idea what that means?

File: prometheus-exporters/image-usage-exporter/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "image-usage-exporter.fullname" . }}
```
Comment (IvoGoman) [inline prometheus-exporters/image-usage-exporter/templates/deployment.yaml:4]: There is no namespace set for the deployment. I would suggest putting it into `infra-monitoring` if there is no other namespace that fits better.

File: prometheus-exporters/image-usage-exporter/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            - name: metrics
+              containerPort: {{ .Values.listen_port }}
+              protocol: TCP
+          env:
+          - name: DB_USER
+            value: {{ .Values.global.datahub.user }}
+          - name: DB_PASSWORD
+            value: {{ .Values.global.datahub.password }}
+          - name: DB_HOST
+            value: "datahubdb-mariadb.datahub.svc.kubernetes.{{ .Values.global.region }}.cloud.sap"
```
Comment (IvoGoman) [inline prometheus-exporters/image-usage-exporter/templates/deployment.yaml:37]: [-] For DB-Host it should be sufficient to use datahubdb-mariadb.datahub.

File: prometheus-exporters/image-usage-exporter/templates/service.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Service
+metadata:
```
Comment (IvoGoman) [inline prometheus-exporters/image-usage-exporter/templates/service.yaml:3]: There are no prometheus scrape annotations.
https://github.com/sapcc/helm-charts/blob/ebeb10415743a2877fe40ea11b236692877ac1bf/prometheus-exporters/elk-query-exporter/templates/service.yaml#L10-L13

**Review by IvoGoman — COMMENTED** [review]
Looks good. But I would set the namespace

**Comment by IvoGoman** [comment]
> `[ERROR] templates/: template: image-usage-exporter/templates/deployment.yaml:33:29: executing "image-usage-exporter/templates/deployment.yaml" at <.Values.global.datahub.user>: nil pointer evaluating interface {}.datahub`
> 
> any idea what that means?

The CI has no access to the secrets. One can set [test values](https://github.com/sapcc/helm-charts/blob/master/prometheus-exporters/elastiflow-query-exporter/ci/test-values.yaml ) so that the CI can render the chart.

**Review by IvoGoman — APPROVED** [review]
